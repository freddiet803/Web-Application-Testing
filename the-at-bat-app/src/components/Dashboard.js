import React from 'react';

 export const Dashboard = props =>{

    // handleStrikeClick = e =>{

    // }

    const handleHitClick = e =>{
       // e.preventDefault();
        props.hits();

    }

    return (
        <div>
            <button onClick={props.strikes}>Strike</button>
            <button onClick={props.balls}>Ball</button>
            <button onClick={props.fouls}>Foul</button>
            <button onClick={handleHitClick}>Hit</button>
            <button onClick={props.reset}>Reset</button>
            
        </div>
    )
}
