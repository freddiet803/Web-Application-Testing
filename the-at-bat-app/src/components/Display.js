import React from 'react';


export const Display = props =>{



    return (
        <div>
            <div>
                <h1>At Bat</h1>
                <div>
                <h3>Balls: {props.batInfo.balls}</h3>
               
                </div>
                <div>
                <h3>Strikes: {props.batInfo.strikes}</h3>
                
                </div>
                <div>
                <h3>Fouls: {props.batInfo.fouls}</h3>
                
                </div>
                <div>
                <h3>Hits: {props.batInfo.hits}</h3>
                
                </div>
                
            </div>

        </div>
    )
}