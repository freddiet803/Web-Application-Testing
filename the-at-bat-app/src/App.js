import React, {useState} from 'react';
import logo from './logo.svg';
import './App.css';
import {Display} from './components/Display.js';
import {Dashboard} from './components/Dashboard.js'

class App extends React.Component {
    // const [atBatInfo, setAtBatInfo] = useState({
    //   balls: 0,
    //   strikes: 0,
    //   hits: 0,
    //   fouls: 0
    // })
    // const incrementChange = e =>{
    //   e.preventDefault();
    //   setAtBatInfo({...atBatInfo,[e.target.name]: e.target.value + 1 });
    // }
    constructor(){
      super();
      this.state= {
        balls: 0,
        strikes: 0, 
        hits: 0, 
        fouls: 0
      }
    }


      increaseBalls = () =>{
        
        this.setState({
          balls: this.state.balls +1
        })
        if(this.state.balls ==4){
          alert('You Walked');
          this.setState({
            
            strikes: 0,
            balls: 0,
            fouls: 0

          })
        }
        console.log(this.state);
      }



  increaseStrikes = () =>{
    
        this.setState({
          strikes: this.state.strikes +1
        });
        if(this.state.strikes == 3){
          alert("Your Out!");
          this.setState({
            
            strikes: 0,
            balls: 0,
            fouls: 0
    
          })
        }
        
        console.log(this.state);
      }
    increaseFouls = () =>{
          
          this.setState({
            fouls: this.state.fouls +1
          });
          if(this.state.strikes <2){
            this.setState({
              strikes: this.state.strikes +1
            });
            
          }
          console.log(this.state);}
     
          increaseHits = () =>{
            
            this.setState({
              hits: this.state.hits +1,
              strikes: 0,
              balls: 0,
              fouls: 0
            });
            console.log(this.state);}

            resetBat = ()=>{
              this.setState({
                hits: 0,
                strikes: 0,
                balls: 0,
                fouls: 0
              });
            }

    

    

    
render(){
  return (
    <div className="App">
      <header className="App-header">
        <Display batInfo={this.state} />
        <Dashboard reset={this.resetBat} hits={this.increaseHits} balls={this.increaseBalls} fouls={this.increaseFouls} strikes={this.increaseStrikes}/>
      </header>
    </div>
  );
}


}


export default App;
