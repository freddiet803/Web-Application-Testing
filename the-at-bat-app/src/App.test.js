import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {render} from '@testing-library/react'
import {Dashboard} from './components/Dashboard.js'
import {Display} from './components/Display.js'
import {shallow, configure} from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// it('renders without crashing', () => {
//   const div = document.createElement('div');
//   ReactDOM.render(<App />, div);
//   ReactDOM.unmountComponentAtNode(div);
// });


test('render without crashing', () =>{
  const container = render(<App/>);
  console.log(container);
  expect(container.firstChild).toMatchSnapshot();
});

test('render dashboard without crashing', ()=>{
  const dashboard = render(<Dashboard/>);
  expect(dashboard.firstChild).toMatchSnapshot();
});

test('render display without crashing', () =>{
  // const theDisplay = render(<Display/>);
  // console.log(theDisplay);

  const wrapper = shallow(
    <Display batInfo={{balls: 0,
      strikes: 0, 
      hits: 0, 
      fouls: 0}}/>
  )
  console.log(wrapper.debug());
  //expect(theDisplay.firstChild).toMatchSnapshot();
    expect(toJson(wrapper)).toMatchSnapshot()
});


