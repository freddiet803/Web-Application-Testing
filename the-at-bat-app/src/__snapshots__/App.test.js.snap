// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render dashboard without crashing 1`] = `undefined`;

exports[`render display without crashing 1`] = `
<div>
  <div>
    <h1>
      At Bat
    </h1>
    <div>
      <h3>
        Balls: 
        0
      </h3>
    </div>
    <div>
      <h3>
        Strikes: 
        0
      </h3>
    </div>
    <div>
      <h3>
        Fouls: 
        0
      </h3>
    </div>
    <div>
      <h3>
        Hits: 
        0
      </h3>
    </div>
  </div>
</div>
`;

exports[`render without crashing 1`] = `undefined`;
